src/MountedTraversal.js

```diff
@@ -161,15 +161,77 @@ export function childrenOfInst(node) {
   return childrenOfInstInternal(internalInstanceOrComponent(node));
 }
 
+// This function should be called with an "internal instance". Nevertheless, if it is
+// called with a "public instance" instead, the function will call itself with the
+// internal instance and return the proper result.
+function findAllInRenderedTreeInternal(inst, test) {
+  if (!inst) {
+    return [];
+  }
+
+  if (!inst.getPublicInstance) {
+    const internal = internalInstance(inst);
+    return findAllInRenderedTreeInternal(internal, test);
+  }
+  const publicInst = inst.getPublicInstance() || inst._instance;
+  let ret = test(publicInst) ? [publicInst] : [];
+  const currentElement = inst._currentElement;
+  if (isDOMComponent(publicInst)) {
+    const renderedChildren = renderedChildrenOfInst(inst);
+    values(renderedChildren || {}).filter((node) => {
+      if (REACT013 && !node.getPublicInstance) {
+        return false;
+      }
+      return true;
+    }).forEach((node) => {
+      ret = ret.concat(findAllInRenderedTreeInternal(node, test));
+    });
+  } else if (
+    !REACT013 &&
+    isElement(currentElement) &&
+    typeof currentElement.type === 'function'
+  ) {
+    ret = ret.concat(
+      findAllInRenderedTreeInternal(
+        inst._renderedComponent,
+        test,
+      ),
+    );
+  } else if (
+    REACT013 &&
+    isCompositeComponent(publicInst)
+  ) {
+    ret = ret.concat(
+      findAllInRenderedTreeInternal(
+        inst._renderedComponent,
+        test,
+      ),
+    );
+  }
+  return ret;
+}
+
+// This function could be called with a number of different things technically, so we need to
+// pass the *right* thing to our internal helper.
```

Comment:
This comment is already in master; I just moved the function higher up due to no-use-before-define